articles/advanced-r-concepts/index.md

```diff
@@ -0,0 +1,333 @@
+2        b        B        2        4     TRUE
+3        c        C        3        6     TRUE
+
+    TRUE TRUE TRUE
+```
+
+The final function under dataframes is `subset.` Subset function is similar to Python's filter function. We can specify conditions using subset and therefore get a slice of the dataframe satisfying the condition. **Let the condition be all those rows where `vector_5` is `not equal` to `FALSE.`**
+
+```r
+subset(new_data_frame, subset= vector_5!=FALSE)
+```
+The output is given as follows:
+
+```txt
+vector_1 vector_2 vector_3 vector_4 vector_5
+1        a        A        1        2     TRUE
+2        b        B        2        4     TRUE
+3        c        C        3        6     TRUE
+```
+
+### Typecasting
+Similar to other programming languages like C++,Python,etc, R provides us with an option to explicitly change or modify the data-type of previosuly defined variables. R provides the keyword as.* keyword to implement the same.
+
+Let's consider the following example:
+
+```r
+x <- 0:6
+class(x)
+typecast_to_numeric <- as.numeric(x)
+class(typecast_to_numeric).0
+typecast_to_logical <- as.logical(x)
+class(typecast_to_logical)
+typecast_to_char <- as.character(x)
+class(typecast_to_char)
+```
+
+The given code outputs the following:
+
+```txt
+> class(x)
+[1] "integer"
+> typecast_to_numeric <- as.numeric(x)
+> class(typecast_to_numeric)
+[1] "numeric"
+> typecast_to_logical <- as.logical(x)
+> class(typecast_to_logical)
+[1] "logical"
+> typecast_to_char <- as.character(x)
+> class(typecast_to_char)
+[1] "character"
+```
+
+Observe the change in the data types. An application of typecasting makes sense on user-defined data-types. Consider the following applications:
+
+1. Consider a dataframe with a column containing boolean values. If one needs to convert these into integers to train a machine learning model, then typecasting is used. Typecasting converts the boolean into integer data.
+
+2. Supposing one is given a dataframe with one column containing time in seconds. To convert each column of the given dataframe to [time in days format](https://www.rdocumentation.org/packages/PivotalR/versions/0.1.18.3.1/topics/Type%20Cast%20functions), explicit typecasting is used.
+
+For more information on explicit typecasting, refer to this [link](https://study.com/academy/lesson/data-type-conversion-in-r-programming-purpose-functions.html).
+
+### Reading and Writing Data
+
+R provides many functions for reading and writing data. There are mainly three data formats used in R. They are `.txt,` `.csv,` and `.RData.` The functions are defined for each of the file-types and are given as follows:
+
+1. read.table: `read.table("test.txt")`
+2. read.csv: `read.csv("test.csv")`
+3. load: `load("file_name.RData")`
+
+For writing into files, R provides the following functions:
+
+1. write.table: `write.table(data, file='file_name.txt')`
+2. write.csv: `write.csv(data, file='file_name.csv')`
+3. save: `save(x,y,"file_name.Rdata")`
+
```

Comment:
Add that this works when the file is in the same directory, otherwise, set up a relative link...
which transitions to your next point